src/java.base/share/classes/java/util/zip/ZipInputStream.java

```diff
@@ -422,7 +422,13 @@ public int read(byte[] b, int off, int len) throws IOException {
         case STORED:
             if (remaining <= 0) {
                 entryEOF = true;
+                final ZipEntry currentEntry = entry;
                 entry = null;
+                if (currentEntry.crc != crc.getValue()) {
+                    throw new ZipException("invalid entry CRC (expected 0x"
+                            + Long.toHexString(currentEntry.crc)
+                            + " but got 0x" + Long.toHexString(crc.getValue()) + ")");
```

Comment:
```suggestion
                    throw new ZipException(""invalid entry CRC (expected 0x%x but got 0x%x)".formatted(currentEntry.crc, crc.getValue()));
```
Can String::formatted be used here?